qadence/ml_tools/train_grad.py

```diff
@@ -125,6 +126,17 @@ def loss_fn(model: torch.nn.Module, data: torch.Tensor) -> tuple[torch.Tensor, d
     # initialize tensorboard
     writer = SummaryWriter(config.folder, purge_step=init_iter)
 
+    perform_val = isinstance(config.val_every, int)
+    if perform_val and not isinstance(dataloader, DictDataLoader):
+        raise ValueError(
+            "If `config.val_every` is provided as an integer, dataloader must"
+            "be an instance of `DictDataLoader`"
+        )
+    if perform_val:
+        iter_keys = list(dataloader.dataloaders.keys())
+        val_dataloader = dataloader.dataloaders[iter_keys[1]]
+        dataloader = dataloader.dataloaders[iter_keys[0]]
+
```

Comment:
I would not by default assume that `val_dataloader` is at index 1. I would rather enforce a key, s.a. `val_dataloader.dataloaders['validation']`

Reply:
Great, thanks! Done.